utils/swift_build_support/swift_build_support/products/product_builder.py

```diff
@@ -0,0 +1,27 @@
+# swift_build_support/product_builders/product_builder.py -------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import abc
+
+
+class ProductBuilder(object):
```

Comment:
This could use a substantial documentation comment 😄

Reply:
I will add something, but I was leaving the docs and test until we are happy with the approach.